src/wallet/sendTransactionWalletService.ts

```diff
@@ -245,9 +251,14 @@ class SendTransactionWalletService extends EventEmitter implements ISendTransact
     }
     this.emit('sign-tx-start');
     const dataToSignHash = this.transaction.getDataToSignHash();
+    const xprivkey = wallet.getXprivKey(this.pin);
 
     for (const [idx, inputObj] of this.transaction.inputs.entries()) {
-      const inputData = this.wallet.getInputData(dataToSignHash, utxosAddressPath[idx]);
+      const inputData = this.wallet.getInputData(
+        xprivkey,
+        dataToSignHash,
+        HathorWalletServiceWallet.getAddressIndexFromFullPath(utxosAddressPath[idx]),
```

Comment:
Add a comment explaining why you need this utility here.

Reply:
Done in 69b03f1f7abd558b9886efee7ef39a0e2ba200e3

Reply:
Great! Thanks!